File: backend/universitet.js
import Emne from "./emne.js";
import Notat from "./notat.js";

export class Universitet {

  constructor(navn, forkortelse) {
    this._navn = navn; //str
    this._forkortelse = forkortelse; //str
    this._emner = {}; //obj, emnekode nøkkel, Emne-objekt innhold
  }

  hentNavn(){
    return this._navn;
  }

  hentForkortelse(){
    return this._forkortelse;
  }

  hentEmne(emnekode){
    return this._emner[emnekode];
  }

  hentEmner(){
    return this._emner;
  }

  leggTilEmne(emne){
    this._emner[emne.hentEmnekode()] = emne;
  }

  toString(){
    return this._forkortelse + ": " + this._navn;
  }

}

File: backend/universitet.py
from emne import Emne

class Universitet:
    def __init__(self, navn, forkortelse):
        self._navn = navn #str
        self._forkortelse = forkortelse #str
        self._emner = {} #emnekode nøkkel, Emne-objekt innhold

    def hentNavn(self):
        return self._navn

    def hentForkortelse(self):
        return self._forkortelse

    def hentEmne(self, emnekode):
        return self._emner[emnekode]

    def hentEmner(self):
        return self._emner

    def leggTilEmne(self, emne):
        self._emner[emne.hentEmnekode()] = emne

    def lastUt(self):
        return self._navn + ";" + self._forkortelse + ";"

    def __str__(self):
        return self._forkortelse + ": " + self._navn
